packages/suite/src/actions/wallet/send/sendFormSolanaActions.ts

```diff
@@ -1,13 +1,145 @@
+
+    if (totalSpent.isGreaterThan(availableBalance)) {
+        const error = token ? 'AMOUNT_NOT_ENOUGH_CURRENCY_FEE' : 'AMOUNT_IS_NOT_ENOUGH';
+        // errorMessage declared later
+        return { type: 'error', error, errorMessage: { id: error } } as const;
+    }
+
+    const payloadData = {
+        type: 'nonfinal',
+        totalSpent: token ? amount : totalSpent.toString(),
+        max,
+        fee: feeInSatoshi,
+        feePerByte: feeLevel.feePerUnit,
+        feeLimit: feeLevel.feeLimit,
+        token,
+        bytes: 0,
+    } as const;
+
+    if (output.type === 'send-max' || output.type === 'external') {
+        return {
+            ...payloadData,
+            type: 'final',
+            // compatibility with BTC PrecomposedTransaction from @trezor/connect
+            transaction: {
+                inputs: [],
+                outputsPermutation: [0],
+                outputs: [
+                    {
+                        address: output.address,
+                        amount,
+                        script_type: 'PAYTOADDRESS',
+                    },
+                ],
+            },
+        };
+    }
+
+    return payloadData;
+};
+
+export const composeTransaction =
+    (formValues: FormState, formState: ComposeActionContext) => () => {
+        const { account, network, feeInfo } = formState;
+        const composeOutputs = getExternalComposeOutput(formValues, account, network);
+        if (!composeOutputs) return; // no valid Output
+
+        const { output, decimals } = composeOutputs;
+
+        // TODO(vl): get custom fee limit from TrezorConnect.BlockchainEstimateFee
+        const customFeeLimit: string | undefined = undefined; // FEE_PER_SIGNATURE.toString();
+
+        // FeeLevels are read-only
+        const levels = customFeeLimit ? feeInfo.levels.map(l => ({ ...l })) : feeInfo.levels;
+        const predefinedLevels = levels.filter(l => l.label !== 'custom');
+        // update predefined levels with customFeeLimit (gasLimit from data size or erc20 transfer)
+        if (customFeeLimit) {
+            predefinedLevels.forEach(l => (l.feeLimit = customFeeLimit));
+        }
+        // in case when selectedFee is set to 'custom' construct this FeeLevel from values
+        if (formValues.selectedFee === 'custom') {
+            predefinedLevels.push({
+                label: 'custom',
+                feePerUnit: formValues.feePerUnit,
+                feeLimit: formValues.feeLimit,
+                blocks: -1,
+            });
+        }
+
+        const wrappedResponse: PrecomposedLevels = {};
+        const response = predefinedLevels.map(level =>
+            calculate(account.availableBalance, output, level),
+        );
+        response.forEach((tx, index) => {
+            const feeLabel = predefinedLevels[index].label as FeeLevel['label'];
+            wrappedResponse[feeLabel] = tx;
+        });
+
+        // format max (calculate sends it as satoshi)
```

Comment:
Satoshi => Lamports

Reply:
Yeah, I've seen that, but I would guess that rather than they prefer it, its just a leftover from times when they had only bitcoin like coins. If its not too much work I would try to stick with "Lamports" where possible